src/Juvix/Compiler/Core/Pretty/Base.hs

```diff
@@ -536,8 +541,59 @@ instance (PrettyCode a) => PrettyCode [a] where
     cs <- mapM ppCode x
     return $ encloseSep "(" ")" ", " cs
 
-{--------------------------------------------------------------------------------}
-{- helper functions -}
+--------------------------------------------------------------------------------
+-- printing values
+--------------------------------------------------------------------------------
+
+instance PrettyCode ConstrApp where
+  ppCode ConstrApp {..} = do
+    n <- ppName KNameConstructor _constrAppName
+    case _constrAppFixity of
+      Nothing -> do
+        args <- mapM (ppRightExpression appFixity) _constrAppArgs
+        return $ hsep (n : args)
+      Just fixity
+        | isBinary fixity ->
+            case _constrAppArgs of
```

Comment:
i'd prefer to have `goBinary`, `goUnary` defined in the where block to improve readability

Reply:
done